### DISCUSSION TITLE
Connecting to a user session on a terminalserver/RD session host

### DISCUSSION THREAD

**Discussion by SvendralRaelis** [rustdesk/rustdesk#10348] [General]
### Bug Description

Connecting to a user session on a terminalserver is behaving weird. The incoming remote window from rustdesk is always appearing on one of the active user sessions which has logged in most recently, directly after establishing a remote connection via the terminalservers rustdesk id. And after selecting someone else on the windows userlist we're getting greeted by a error message "connection error" and afterwards "wrong password", so I have to try it again with the newly generated password. Only then the remote connection to a terminalserver user works.  

### How to Reproduce

Connect to a windows machine with multiple active user sessions (like on a terminalserver/RD session host) using the rustdesk ID of that machine.
The incoming remote window from rustdesk will appear on the screen of one of the user sessions.
Choose any other user instead of the user with that rustdesk window open and you'll get greeted by that error message I've mentioned.

### Expected Behavior

After connecting to the "terminalserver/RD session host/windows machine with multiple active user sessions" the rustdesk incoming remote window shouldn't appear before you've selected any user from the user sessions list. Then you can either input the permanent password from the rustdesk host, or the user on the terminalserver opens rustdesk and can tell you the temporary password or just accept the remote connection. Then the remote session gets established. 

### Operating system(s) on local (controlling) side and remote (controlled) side

Windows 11 -> Windows Server 2019 

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.2.6 -> 1.2.6

### Screenshots

[https://www.youtube.com/watch?v=pq1eCjuIQYI](https://www.youtube.com/watch?v=pq1eCjuIQYI)
![image](https://github.com/user-attachments/assets/f4396cc3-cf15-44a0-b77a-18b7c21e35d5)

**Comment by rustdesk** [comment]
@21pages low priority

**Comment by SvendralRaelis** [comment]
I figured out that apparently on a terminalserver that remote connection incoming popup window alwas appears first either on the last logged in user of that terminalserver, or in case there was already a rustdesk remote connection session established with an user on that terminalserver, the incoming remote connection popup will always appear for that user.
On that note, for a workaround, does someone know where rustdesk saves the last session/s? Is it an actual file or a registry key? I couldn't find anything so far myself.
